Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -82,25 +85,66 @@ public class DiagnosticErrorReporter : ErrorReporter {
       );
     }
 
-    private static IEnumerable<DiagnosticRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string message) {
+    public static readonly string PostConditionFailingMessage = new EnsuresDescription().FailureDescription;
+    private readonly string entryDocumentsource;
+
+    public static string FormatRelated(string related) {
+      return $"Could not prove: {related}";
+    }
+
+    private IEnumerable<DiagnosticRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string message) {
+      var (tokenForMessage, inner) = token is NestedToken nestedToken ? (nestedToken.Outer, nestedToken.Inner) : (token, null);
+      if (tokenForMessage is RangeToken range) {
+        var rangeLength = range.EndToken.pos + range.EndToken.val.Length - range.StartToken.pos;
+        if (message == PostConditionFailingMessage) {
+          var postcondition = entryDocumentsource.Substring(range.StartToken.pos, rangeLength);
+          message = $"This postcondition might not hold: {postcondition}";
+        } else if (message == "Related location") {
+          // We can do better than that.
+          var tokenUri = tokenForMessage.GetDocumentUri();
+          var tokenUriSystemPath = tokenUri.GetFileSystemPath();
+          var entryDocumentUriSystemPath = entryDocumentUri.GetFileSystemPath();
+          if (tokenUri == entryDocumentUri || tokenUriSystemPath == entryDocumentUriSystemPath ||
+              tokenUriSystemPath == "/" + entryDocumentUriSystemPath) {
```

Comment:
Oof. Is there a way we can normalize the system path to avoid having to check all of these cases?

Reply:
Oh right. I had been fighting to enable a simpler code but I forgot this case. Let me fix it.